woocommerce/woocommerce-ios
# Adds SurveyCoordinatingController

closes #2544 

# Why

This PR adds the navigation wires for the in-app feedback survey feature as discussed in p91TBi-36A-p2
Huge thanks to @jaclync @pmusolino, @shiki, and @jleandroperez for all the input.

# How

- Introduced `SurveyCoordinatingController`, a `NavigationController` subclass which is in charge of deciding navigation between its view controllers.
- Introduced `ViewControllersFactory` protocol and `ViewControllerOutputs` protocols to decouple and inject view controller creation - Needed for tests -
- Add tests for the navigation flow 💪 

# Gif
![SurveyNavigation](https://user-images.githubusercontent.com/562080/89535934-bc3a9200-d7bc-11ea-9719-81f43deaedee.gif)

# Testing Steps
- Launch the app and tap on the "Could be better button" on the dashboard
- Complete the survey
- See that the app navigates to the "Feedback Sent" screen
- Tap on "contact us" and see that you can create a zendesk issue
- Tap on "back to store" and the that the flow is dismissed

Update release notes:
